controller/src/main/java/io/pravega/controller/store/stream/tables/TableHelper.java

```diff
@@ -693,102 +631,53 @@ public static boolean isRerunOf(final List<Integer> segmentsToSeal,
     }
 
     /**
-     * It finds the segment sealed event between lower and upper where 'lower' offset is guaranteed to be greater than or equal to segmentCreatedEvent
-     * @param lower starting record number in index table from where to search
-     * @param upper last record number in index table till where to search
+     * It finds the segment sealed event between lower and upper where 'lower' offset is guaranteed to be greater than or
```

Comment:
This seems to be one use for segment creation epoch, are there other uses for creation epoch?

Reply:
Earlier we used to perform binary search over history table to find segment creation epoch in history table. 
It is easier to keep the epoch number as part of segment record so that we can do O(1) access. 
Segment creation epoch is used in methods for finding successors and predecessors.

Reply:
I am making the change and adding corresponding test.. thanks